src/dmd/cparse.d

```diff
@@ -5078,4 +5100,63 @@ final class CParser(AST) : Parser!AST
     }
 
     //}
+
+    /******************************************************************************/
+    /********************************* #define Parser *****************************/
+    //{
+
+    /**
+     * Go through the #define's in the defines buffer and see what we can convert
+     * to Dsymbols, which are then appended to symbols[]
+     */
+    void addDefines()
+    {
+        if (!defines || defines.length < 10)  // minimum length of a #define line
+            return;
+        const length = defines.length;
+        auto slice = defines.peekChars()[0 .. length];
+        resetDefineLines(slice);                // reset lexer
+
+        const(char)* endp = &slice[length - 7];
+
+        Token n;
+
+        while (p < endp)
+        {
+            if (p[0 .. 7] == "#define")
+            {
+                p += 7;
+                scan(&n);
+                //printf("%s\n", n.toChars());
+                if (n.value == TOK.identifier)
+                {
+                    auto id = n.ident;
+                    scan(&n);
+                    if (n.value == TOK.int32Literal)
```

Comment:
This entire PR will just end up being reverted when we start doing the more complex cases, might as well do it right from the start.

Reply:
It'll still need the rest of the PR. The #defines still have to be captured.